cmd/bnsd/app/app.go

```diff
@@ -38,7 +39,7 @@ func Authenticator() x.Authenticator {
 
 // Chain returns a chain of decorators, to handle authentication,
 // fees, logging, and recovery
-func Chain(authFn x.Authenticator) app.Decorators {
+func Chain(authFn x.Authenticator, minFee coin.Coin) app.Decorators {
```

Comment:
Yes, this is good.

You can optionally replace the handler with nil if minFee is zero, but that is an optimizaton

Reply:
@ethanfrey this does not work for some reason, scenarios fail when I have this condition there. So I'll keep it simple for now.